token-swap/program/src/processor.rs

```diff
@@ -992,6 +999,52 @@ impl Processor {
         Ok(())
     }
 
+    /// Processes a [ToggleSwap](enum.Instruction.html).
+    pub fn process_toggle_swap(
+        is_trading: bool,
+        fees: Fees,
+        swap_curve: SwapCurve,
+        accounts: &[AccountInfo],
+    ) -> ProgramResult {
+        let account_info_iter = &mut accounts.iter();
+        let swap_info = next_account_info(account_info_iter)?;
+        let trading_authority = next_account_info(account_info_iter)?;
+
+        if !trading_authority.is_signer {
+            return Err(ProgramError::MissingRequiredSignature);
+        }
+
+        let token_swap = SwapVersion::unpack(&swap_info.data.borrow())?;
+
+        if trading_authority.key != token_swap.trading_authority() {
+            return Err(ProgramError::InvalidAccountData);
+        }
+
+        if *token_swap.fees() != fees || *token_swap.swap_curve() != swap_curve {
+            return Err(SwapError::IncorrectSwapAccount.into());
+        }
+
+        let obj = SwapVersion::SwapV1(SwapV1 {
+            is_initialized: token_swap.is_initialized(),
+            nonce: token_swap.nonce(),
+            token_program_id: *token_swap.token_program_id(),
+            token_a: *token_swap.token_a_account(),
+            token_b: *token_swap.token_b_account(),
+            pool_mint: *token_swap.pool_mint(),
+            token_a_mint: *token_swap.token_a_mint(),
+            token_b_mint: *token_swap.token_b_mint(),
+            pool_fee_account: *token_swap.pool_fee_account(),
+            fees,
+            swap_curve,
+            is_trading: is_trading,
+            trading_authority: *token_swap.trading_authority(),
+        });
+
+        SwapVersion::pack(obj, &mut swap_info.data.borrow_mut())?;
```

Comment:
alternatively, you could add another unpack function for `SwapVersion` in `state.rs`.
```
pub fn unpack_as_version(input: &[u8]) -> Result<SwapVersion, ProgramError> {
        let (&version, rest) = input
            .split_first()
            .ok_or(ProgramError::InvalidAccountData)?;
        match version {
            1 => Ok(SwapVersion::SwapV1(SwapV1::unpack(rest)?)),
            _ => Err(ProgramError::UninitializedAccount),
        }
}
```
then in processor
```
match &mut token_swap {
            SwapVersion::SwapV1(v) => v.is_trading = is_trading
};

SwapVersion::pack(token_swap, &mut swap_info.data.borrow_mut())?;
```

this is probably the better version because it will stop functioning as soon as another version is added in `state` because `match` needs to be exhaustive

Reply:
Done.